### PR TITLE
PSoC64 Release Branch to Master

### PR DESCRIPTION
**Pull request by ifyall**
CY8CKIT-064S0S2-4343W Production Updates

Description
-----------
Update CY8CKIT-064S0S2-4343W to cysecuretools v3.1 and newest versions of kit firmware. Update Cypress configuration files to align to new Amazon FreeRTOS standards. Updated OTA support to new version of mcuboot and added support for the SWAP feature.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mingyue86010** [comment]
rebased to top of main branch.

**Review by mingyue86010 — DISMISSED** [review]

**Comment by mingyue86010** [comment]
This PR contains some changes to FreeRTOS_CLI and Espressif demo code which shouldn't be. Those changes need to be remove from this PR.

**Review by mingyue86010 — APPROVED** [review]

File: vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_tests/config_files/iot_config.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ extern void IotBleMqtt_CleanupSerialize( void );
 #define IOT_HTTPS_DISPATCH_USE_STATIC_MEMORY    1
 #define IOT_HTTPS_DISPATCH_TASK_STACK_SIZE      ( configMINIMAL_STACK_SIZE * 4 )
 
-#define BLE_SUPPORTED                           ( 1 )
+//#define BLE_SUPPORTED                           ( 1 )
```
Comment (paulbartell) [inline vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_tests/config_files/iot_config.h:69]: Should this be commented out?

File: vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_tests/config_files/aws_ota_agent_config.h [hunk 2]
```diff
@@ -60,7 +58,7 @@
  * @brief The OTA agent task priority. Normally it runs at a low priority.
  *    lowest: 0, max 7 (configMAX_PRIORITIES) (tskIDLE_PRIORITY == 0)
  */
-#define otaconfigAGENT_PRIORITY                 (2)
+#define otaconfigAGENT_PRIORITY                 (0)
```
Comment (paulbartell) [inline vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_tests/config_files/aws_ota_agent_config.h:61]: Is this intentional?

File: vendors/cypress/MTB/port_support/ota/mcuboot/cy_flash_pal/cy_flash_map.c [hunk 3]
```diff
@@ -132,14 +132,20 @@ static struct flash_area secondary_1 =
     .fa_id = FLASH_AREA_IMAGE_SECONDARY(0),
```
Comment (paulbartell) [inline vendors/cypress/MTB/port_support/ota/mcuboot/cy_flash_pal/cy_flash_map.c:132]: Could mcuboot be integrated as a submodule instead?

File: vendors/cypress/MTB/psoc6/cmake/cy_create_linked_target.cmake [hunk 4]
```diff
@@ -155,7 +155,7 @@ function(cy_add_link_libraries)
         TOOLCHAIN     "${AFR_TOOLCHAIN}"
```
Comment (paulbartell) [inline vendors/cypress/MTB/psoc6/cmake/cy_create_linked_target.cmake:155]: Is a reason we have both cmake and make build files for this port?

File: vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/aws_bufferpool_config.h [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
```
Comment (paulbartell) [inline vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/aws_bufferpool_config.h:1]: Was this file removed on purpose?

File: vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/aws_ota_agent_config.h [hunk 6]
```diff
@@ -60,7 +58,7 @@
  * @brief The OTA agent task priority. Normally it runs at a low priority.
  *    lowest: 0, max 7 (configMAX_PRIORITIES) (tskIDLE_PRIORITY == 0)
  */
-#define otaconfigAGENT_PRIORITY                 (2)
+#define otaconfigAGENT_PRIORITY                 (0)
```
Comment (paulbartell) [inline vendors/cypress/boards/CY8CKIT_064S0S2_4343W/aws_demos/config_files/aws_ota_agent_config.h:61]: Is this change intentional?

**Review by paulbartell — APPROVED** [review]
A few questions, but looks good overall.

**Review by n9wxu — APPROVED** [review]
